### ISSUE TITLE
Why does Safari hate Vue?

### ISSUE THREAD

**Issue by guyewhite** [susanBuck/e28-spring20#50]
Hi Team,

Without critiquing my choice in browsers, why is it that Safari seems to act so finicky related to Vue? For example, as I'm building ZipFoods, the hot loading does not seem to be happening. When doing some of my other projects related to this class, I often see unpredictable behavior in Safari. This, naturally, concerns me a bit as about 40% of my users in one of my businesses are Safari users. I've Googled the topic, but have not found many clues. Thanks for your input.

**Comment by susanBuck** [comment]
Hi @guyewhite - 

Vue.js supports all browsers that are ES5-compliant, which - if we exclude obsolete versions - Safari is ([ref](http://kangax.github.io/compat-table/es5/)).

Given this, all functionality from the user's perspective *should* work the same in Safari as it does on Firefox, Chrome, etc.

I italicize *should* because sometimes browser compatibility in theory vs in practice can be very different things. : ) 

**Can you recall any other specifics of when you've seen finicky/unpredictable behavior?** Being able to troubleshoot/look at a more concrete example would be helpful to understand what's going on, if it is browser specific, etc.

**On the topic of hot module reloading in Safari** - 
I just tested it on my end and found that Safari did reload. However, there is a weird bug where if I manually refresh the page it reverts to what the page looked like before the HMR did the refresh.

I found a thread on this and implemented [the solution described here](https://github.com/vuejs/vue-cli/issues/1132#issuecomment-409916879) and that seemed to fix it.

But in your case, HMR is not working at all in Safari, yes? Just out of curiosity, can you confirm that it is working in another browser such as Chrome?

**Comment by guyewhite** [comment]
Hi @susanBuck!

The place where this was most pronounced between time indexes 9:30 and 12:30 in the Week 9 Part 3 video. After getting all that code and those assets in place, the page (even with reload) was still only showing the default Vue page.

Reading the linked page you provided, that surely fits what I am seeing.

As we discuss this, I'm realizing the unpredictable behavior definitely takes place only when I'm developing the application. After a clear browser cache and a completed project, all seems to be working well. Hence, my workflow has been to develop using Chrome and then only test the final product with a martini in hand with Safari.

Chrome ALWAYS works while in development.

**Comment by susanBuck** [comment]
@guyewhite - Thanks for those extra details! I'll take a look at that part of the video again and run some tests in Safari. Will follow-up with what I find. 👍

**Comment by susanBuck** [comment]
I just went through a quick new project set up using Safari. Per the thread I linked above, I added a `vue.config.js` file in the root with this code:

```
module.exports = {
  chainWebpack: config => {
    if (process.env.NODE_ENV === 'development') {
      config
        .output
        .filename('[name].[hash].js') 
        .end() 
    }  
  }
}
```

All seemed to work well, even HMR. Curious what your experience in Safari is like after you add that "fix". 

If it doesn't work, I support the "build in Chrome, test in Safari with a martini" approach. :-)
